### PR TITLE
Initialize klog in clusterctl

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@vincepri.com>

**What this PR does / why we need it**:
This PR fixed klog initialization in clusterctl by copying the glog flags. This should be only needed temporarily while upstream dependencies are still depending on glog.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/clusterctl/cmd/root.go [hunk 1]
```diff
@@ -30,6 +30,21 @@ var RootCmd = &cobra.Command{
 	Use:   "clusterctl",
 	Short: "cluster management",
 	Long:  `Simple kubernetes cluster management`,
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		// This is a temporary hack to enable proper logging until upstream dependencies
+		// are migrated to fully utilize klog instead of glog.
```
Comment (justinsb) [inline cmd/clusterctl/cmd/root.go]: This is a great approach (I had written my own hack, but this is better) 👍

File: cmd/clusterctl/cmd/root.go [hunk 2]
```diff
@@ -30,6 +30,21 @@ var RootCmd = &cobra.Command{
...
+		klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+		klog.InitFlags(klogFlags)
+
+		// Sync the glog and klog flags.
+		cmd.Flags().VisitAll(func(f1 *pflag.Flag) {
```
Comment (justinsb) [inline cmd/clusterctl/cmd/root.go]: I'm actually wondering if we should put this as a helper function into klog ... it seems broadly applicable!

**Comment by justinsb** [comment]
Not sure I agree that this is going to be _that_ short-lived, but this is a really nice approach.

/lgtm

**Comment by vincepri** [comment]
Fixed tests 🎉

**Comment by roberthbailey** [comment]
@dims - can you confirm that this looks reasonable to you? And also consider whether we should move some of this to klog, since it feels like many projects are going to be in a state where they vendor k8s deps that use glog for quite a while.

**Comment by flaper87** [comment]
This seems to fix the issue: Should we do the same for the manager?

/approve

**Comment by dims** [comment]
@roberthbailey this looks reasonable especially because we end up with `logtostderr` set to `true` in one of the `init` methods so there are no files written to disk. If that were not the case there would be two entities that would write to disk (one from glog and one from klog). So it would be best to keep it this way and also over a period of time switch over more and more of the components to klog.

/lgtm

**Comment by dims** [comment]
Added a reminder so i won't forget https://github.com/kubernetes/klog/issues/26

**Pull request merged**